docs/Customization/Heterogeneous-SoCs.rst

```diff
@@ -100,27 +84,20 @@ Named ``MultiRoCCKey``, this key allows you to attach RoCC accelerators based on
 For example, using this allows you to create a 8 tile system with a RoCC accelerator on only a subset of the tiles.
 An example is shown below with two BOOM cores, and one Rocket tile with a RoCC accelerator (Hwacha) attached.
 
-.. code-block:: scala
-
-    class DualBoomAndOneHwachaRocketConfig extends Config(
-      new WithTop ++
-      new WithBootROM ++
-      new WithMultiRoCC ++
-      new WithMultiRoCCHwacha(0) ++ // put Hwacha just on hart0 which was renumbered to Rocket
-      new boom.system.WithRenumberHarts(rocketFirst = true) ++
-      new hwacha.DefaultHwachaConfig ++
-      new boom.common.WithRVC ++
-      new boom.common.LargeBoomConfig ++
-      new boom.system.WithNBoomCores(2) ++
-      new freechips.rocketchip.subsystem.WithoutTLMonitors ++
-      new freechips.rocketchip.subsystem.WithNBigCores(1) ++
-      new freechips.rocketchip.system.BaseConfig)
+.. literalinclude:: ../../generators/example/src/main/scala/HeteroConfigs.scala
+    :language: scala
+    :start-after: DOC include start: DualBoomAndRocketOneHwacha
+    :end-before: DOC include end: DualBoomAndRocketOneHwacha
 
 In this example, the ``WithRenumberHarts`` relabels the ``hartId``'s of all the BOOM/Rocket cores.
-Then after that is applied to the parameters, the ``WithMultiRoCCHwacha(0)`` is used to assign to ``hartId`` zero a Hwacha (in this case ``hartId`` zero is Rocket).
+Then after that is applied to the parameters, the ``WithMultiRoCCHwacha`` is used to assign to a Hwacha accelerator to a particular ``hartId`` (in this case the ``hartId`` of ``2`` corresponds to the Rocket core).
 Finally, the ``WithMultiRoCC`` mixin is called.
 This mixin sets the ``BuildRoCC`` key to use the ``MultiRoCCKey`` instead of the default.
 This must be used after all the RoCC parameters are set because it needs to override the ``BuildRoCC`` parameter.
 If this is used earlier in the configuration sequence, then MultiRoCC does not work.
 
 This mixin can be changed to put more accelerators on more cores by changing the arguments to cover more ``hartId``'s (i.e. ``WithMultiRoCCHwacha(0,1,3,6,...)``).
+
+
+.. [1] Note, in this section core and tile are used interchangeably but there is subtle distinction between a core and tile (tile contains a core, L1D/I$, PTW).
```

Comment:
Maybe add quotes around "core" and "tile"?

Reply:
Done